falcon/util/uri.py

```diff
@@ -178,9 +178,8 @@ def decode(encoded_uri):
         tokens = decoded_uri.split('%')
         decoded_uri = tokens[0]
         for token in tokens[1:]:
-            char, byte = _HEX_TO_BYTE[token[:2]]
-            decoded_uri += char + token[2:]
-
+            char, byte = _HEX_TO_BYTE.get(token[:2], ('%', 0))
+            decoded_uri += char + (token[2:] if byte else token)
```

Comment:
This topic was kindly discussed at #637. I quote "A `try/except` block is extremely efficient if no exceptions are raised Actually catching an exception is expensive". Since there are chances that `token[:2]` can be anything, I conceive we can use `if ... in` here instead of `try...except KeyError`. The `if ... in` block, which is also more efficient than `dict.get()`, can balance the time cost better than `try/except` block.